clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/AbstractProtocolConfigurationBuilder.java

```diff
@@ -57,14 +57,18 @@
  */
 public abstract class AbstractProtocolConfigurationBuilder<P extends Protocol, C extends ProtocolConfiguration<P>> implements ResourceServiceBuilder<C>, Value<C>, ProtocolConfiguration<P>, Consumer<P> {
 
+    private final String stack;
     private final String name;
     private final InjectedValue<ModuleLoader> loader = new InjectedValue<>();
     private final InjectedValue<ProtocolDefaults> defaults = new InjectedValue<>();
+    private final InjectedValue<Boolean> stackStatisticsEnabled = new InjectedValue<>();
 
     private volatile Map<String, String> properties;
     private volatile String moduleName;
+    private volatile ModelNode statisticsEnabled;
 
-    protected AbstractProtocolConfigurationBuilder(String name) {
+    protected AbstractProtocolConfigurationBuilder(String stack, String name) {
```

Comment:
The additional constructor parameter is unnecessary.  Only values that a needed to construct the service name should appear here.  The stack name can be obtained in configure(...) from OperationContext.getCurrentAddress().getParent().getLastElement().getValue().